src/lib/persistence.ts

```diff
@@ -0,0 +1,184 @@
+import type { BaseTable } from "@/tables/types";
+import { getEntitySymbol, type Entity, type EntitySymbol } from "@/lib/external/mud/entity";
+import type { MappedType, Primitive, BaseTableMetadata, Properties, Schema } from "@/lib/external/mud/schema";
+import { Type } from "@/lib/external/mud/schema";
+
+/* -------------------------------------------------------------------------- */
+/*                                    TYPES                                   */
+/* -------------------------------------------------------------------------- */
+
+/**
+ * {
+ *   "RETA_tables_0x0000...127": {
+ *     x: {
+ *       "0xabc...def": 10,
+ *       "0x123...456": 5
+ *     },
+ *     y: {
+ *       "0xabc...def": 20,
+ *       "0x123...456": 15
+ *     },
+ *     ...
+ *   },
+ *   "RETA_tables_Score": {
+ *     ...
+ *   },
+ * }
+ */
+type Serializable = Serialized | Array<Serialized>;
+type Serialized = string | number | boolean;
+type StoredTable<PS extends Schema> = { [key in keyof PS]: { [entity: Entity]: Serialized } };
+
+/* -------------------------------------------------------------------------- */
+/*                                   STORAGE                                  */
+/* -------------------------------------------------------------------------- */
+
+const LOCAL_STORAGE_KEY = "RETA";
+const PREFIX = `${LOCAL_STORAGE_KEY}_tables_`;
+const getStorageKey = (tableId: string) => `${PREFIX}${tableId}`;
+
+const getStoredTable = <PS extends Schema>(tableId: string) => {
+  const storageKey = getStorageKey(tableId);
+  const storedTable = localStorage.getItem(storageKey);
```

Comment:
This will break in non-node environments. Is there a way we can pass in a custom storage object so it works? Or else throw an error in node env?

Reply:
Environment is handled at `createTable`: 
```ts
if (persist && typeof window === "undefined")
  throw new Error("Tables cannot be persisted in a non-browser environment");
```

So it should throw before it attempts to call `getStoredTable` or any of such methods.

By custom storage object, does it mean passing a custom object with the same methods (`getAllProperties`, `setProperties`, `updateProperties`) to `createWrapper` which, if provided, gets used instead to access/update supposed local storage?